docs/IsolationNetworkPolicies.md

```diff
@@ -0,0 +1,105 @@
+# Isolation 
+
+_**Note:**_ below we start with Network Polices as an isolation method, later we will add another methods.
+
+## Introduction
+
+In the current Fybrik implementation the entry point of a data plane is published as a part of the FybrikApplication 
+status, it is completely non-protected, and as result everyone can connect to the data plane. 
+In “Read” flow cases it leads to governance/security violations, “Write” and “Delete” flows can create severe accidents, 
+such as unauthorized data changes or deletions. Given that we need to prevent unauthorized data plane access. 
+
+_Note:_ the same situation exists between chained modules.  
+
+## Challenge
+
+When we talk about accessing Fybrik data plane, we are talking about connectivity between a user workload, which is out of 
+Fybrik scope, and Fybrik module, which can be developed by a third party. Therefore, in this document (and in its 
+extensions) we provide our recommendations how to protect data plane access and possible solutions.Responsibility of the
+data plane isolation is shared between Fybrik core, Fybrik users, module developers and probably IT teams.
+
+- Fybrik users should provide all relevant information to Fybrik and correctly configure their workloads. For example, 
+to set labels on the workload and inform Fybrik about the labels.
+
+- Depends on the isolation method, Fybrik should propagate all relevant information to the deployed modules, and/or 
+correctly configure Kubernetes Network Policies.
+- Module developers should allow to protect the module instances. For example, to allow TLS support
+
+In some complicated cases, IT teams interfere can be required, e.g. to configure external access points.
+
+## Requirements
+
+- Only predefined users/workloads should be able to access the data plane.
+- We do not assume that client workloads will be collocated on the same Kubernetes cluster with the data module, 
+furthermore they can run out of Kubernetes at all. 
+  - However, we do check the collocation of a Fybrik module and a relevant workload in the same cluster as a specific use case. 
+- We should allow workload client connections from different IP addresses. 
+
+
+## Network Policies
+
+Fybrik is a Kubernetes application, therefore, the simplest isolation method can be based on 
```

Comment:
What are the other options, other than network policies?  Why were they ruled out?

Reply:
I plan to work incrementally, they will be added later in parallel with NP implementation

Reply:
If there is another design doc mentioning all the approaches please link to it.

Reply:
It is in a box under development now. After that it will be a separate document or extension of this one.